lib/membrane_http_adaptive_stream/hls_source.ex

```diff
@@ -0,0 +1,164 @@
+defmodule Membrane.HLS.Source do
+  use Membrane.Source
+  require Membrane.Pad, as: Pad
+
+  alias __MODULE__.ClientGenServer
+
+  alias Membrane.{
+    AAC,
+    H264,
+    RemoteStream
+  }
+
+  def_output_pad :video_output,
+    accepted_format: any_of(H264, %RemoteStream{content_format: H264}),
+    availability: :on_request,
+    max_instances: 1,
+    flow_control: :manual,
+    demand_unit: :buffers
+
+  def_output_pad :audio_output,
+    accepted_format: any_of(AAC, %RemoteStream{content_format: AAC}),
+    availability: :on_request,
+    max_instances: 1,
+    flow_control: :manual,
+    demand_unit: :buffers
+
+  def_options url: [spec: String.t()],
+              buffer_size: [spec: pos_integer(), default: 0],
+              container: [spec: :mpeg_ts | :fmp4, default: :mpeg_ts]
```

Comment:
I don't think we need to explicitly specify `container`, ExHLS should be able to automatically resolve it based on the extension of the segments.